.github/workflows/push.yaml

```diff
@@ -0,0 +1,93 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+# This workflow will build a Java project with Gradle and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-java-with-gradle
+
+name: Publish Polaris Server
+
+on:
+  push:
+    branches: [ "GH-635-add-build-workflow" ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+
+    steps:
+      - uses: actions/checkout@v4
+      # Get the current date in YYYY-mm-dd format to be used as the image tag
+      - name: Get current date
+        id: date
+        run: echo "date=$(date +'%Y-%m-%d')" >> $GITHUB_OUTPUT
+      - name: Set up JDK 21
+        uses: actions/setup-java@v4
+        with:
+          java-version: '21'
+          distribution: 'temurin'
+
+      # Configure Gradle for optimal use in GiHub Actions, including caching of downloaded dependencies.
+      # See: https://github.com/gradle/actions/blob/main/setup-gradle/README.md
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@16bf8bc8fe830fa669c3c9f914d3eb147c629707 # v4
+        with:
+          # The setup-gradle action fails, if the wrapper is not using the right version or is not present.
+          # Our `gradlew` validates the integrity of the `gradle-wrapper.jar`, so it's safe to disable this.
+          validate-wrappers: false
+
+      - name: Build with Gradle Wrapper
+        run: ./gradlew build
+        env:
+          ORG_GRADLE_PROJECT_mavenUser: ${{ secrets.JFROG_USER }}
+          ORG_GRADLE_PROJECT_mavenPassword: ${{ secrets.JFROG_TOKEN }}
+          ORG_GRADLE_PROJECT_snapshotsRepoURL: ${{ vars.MAVEN_SNAPSHOT_URL }}
+          ORG_GRADLE_PROJECT_releasesRepoURL: ${{ vars.MAVEN_RELEASE_URL }}
+      # The following step is needed to enable multi-platform Docker builds  
+      - uses: docker/setup-buildx-action@v3
```

Comment:
`docker/setup-qemu-action@v3` is also needed for some architectures in multi-platform builds. https://docs.docker.com/build/ci/github-actions/multi-platform/